CRM/Membershipextra/Utils.php

```diff
@@ -35,49 +41,51 @@ public static function setSetting($value, $name, $typeID) {
    */
   public static function getMembershipTypeConfiguredonForm($form) {
     $formMembershipTypeIds = $membershipExtras = $groupConfigured = [];
-    foreach ($form->_values['fee'] as $fee_id => $fee) {
-      if (!is_array($fee['options']))
+    foreach ($form->_values['fee'] as $fee) {
+      if (!is_array($fee['options'])) {
         continue;
-      foreach ($fee['options'] as $option_id => &$option) {
-        // if priset contain the membership type, then proceed.
+      }
+      foreach ($fee['options'] as $option) {
+        // if priceset contain the membership type, then proceed.
         if (isset($option['membership_type_id'])) {
-          $formMembershipTypeIds[$option_id] = $option['membership_type_id'];
+          $membershipTypeId = $option['membership_type_id'];
+          $formMembershipTypeIds[$option['price_field_id']] = $membershipTypeId;
```

Comment:
in _validation hook are using price field options and here you are passing price field id, so validation get failed.
After applying following changes on top of your changes.. it start showing validation if somebody try to renew their membership before allowed time in both rolling and Fixed Membership type case.

```patch
diff --git a/CRM/Membershipextra/Utils.php b/CRM/Membershipextra/Utils.php
index f22f271..9653f30 100644
--- a/CRM/Membershipextra/Utils.php
+++ b/CRM/Membershipextra/Utils.php
@@ -45,11 +45,11 @@ class CRM_Membershipextra_Utils {
       if (!is_array($fee['options'])) {
         continue;
       }
-      foreach ($fee['options'] as $option) {
+      foreach ($fee['options'] as $option_id => $option) {
         // if priceset contain the membership type, then proceed.
         if (isset($option['membership_type_id'])) {
           $membershipTypeId = $option['membership_type_id'];
-          $formMembershipTypeIds[$option['price_field_id']] = $membershipTypeId;
+          $formMembershipTypeIds[$option_id] = $membershipTypeId;
 
           // get Custom Setting for each Membership type
           $membershipExtras[$membershipTypeId] = CRM_Membershipextra_Utils::getSettings($membershipTypeId);
```